tests/test_join.py

```diff
@@ -0,0 +1,153 @@
+import os
+
+import requests
+from pprint import pprint
+
+
+class BalanceTests:
+    """
+    Unit tests for the Back-End.
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """
+        This method is run once before any of the class' test methods are run.
+        It sets up the environment for the tests, and checks that the token is provided and the API is up.
+        """
+
+        cls.base_url = 'http://localhost:5000'
```

Comment:
is there any way to re-use some of this code that's duplicated in the other file?
Perhaps creating a file for these repeated functions that returns a tuple of the exact dependencies that need to be set on the cls method?

Reply:
scope creep, seperate issue made here https://cs401.atlassian.net/browse/BE-108

Reply:
not sure what u mean by scope creep but avoiding the introduction of duplicated code is just following good programming practices for a solid implementation

Reply:
Refactoring these setup calls is unrelated to testing joining groups. there is a separate issue that @tdbucks is currently working on that is a refactor of the tests and will contain this fix

Reply:
Sounds good. If this is within the scope of a pre-existing issue, do we still need BE-108? At this point in time I say we should avoid creating more issues unless absolutely necessary